compiler-core/src/type_/expression.rs

```diff
@@ -691,6 +691,27 @@ impl<'a, 'b> ExprTyper<'a, 'b> {
                 unify(left.type_(), right.type_())
                     .map_err(|e| convert_unify_error(e, right.location()))?;
 
+                match (&left, &right) {
+                    (TypedExpr::Call { fun, .. }, TypedExpr::Int { value, .. })
+                    | (TypedExpr::Int { value, .. }, TypedExpr::Call { fun, .. }) => {
+                        match fun.as_ref() {
+                            TypedExpr::ModuleSelect {
+                                module_name, label, ..
+                            } => {
+                                if module_name == "gleam/list" && label == "length" && value == "0"
```

Comment:
A general question I have is whether this is the best way to detect the usage of `list.length`.

It does feel a little weird since the compiler doesn't know about the standard library (AFAIK).